numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -2123,7 +2123,7 @@ getset_numericops(PyObject* NPY_UNUSED(self), PyObject* NPY_UNUSED(args))
 static PyObject *
 uses_new_casts(PyObject* NPY_UNUSED(self), PyObject* NPY_UNUSED(args))
 {
-#if NPY_USE_NEW_CASTINGIMPL
+#ifdef NPY_USE_NEW_CASTINGIMPL && NPY_USE_NEW_CASTINGIMPL > 0
```

Comment:
would have to be `#if defined(...) && ...` but as I said, I think the global solution is better.